cmd/juju/cloud/add.go

```diff
@@ -20,6 +20,14 @@ import (
 	"github.com/juju/juju/environs"
 )
 
+type CloudQueryer interface {
+	ParseCloudMetadataFile(path string) (map[string]cloud.Cloud, error)
```

Comment:
But we actually do need to test that the interactive add cloud code *actually* produces valid yaml that can be parsed.  If we mock that out, then we don't know.  I could go change what the parsing code expected, and these tests would all pass, but the code would be wrong and would not work in production.